src/redux/pattern/pattern.sagas.js

```diff
@@ -0,0 +1,26 @@
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { setPatterns } from './pattern.actions';
+
+import { getAllPatterns } from './pattern.operations';
+
+import { GET_PATTERNS } from './pattern.types';
+import { setError } from '../error/error.actions';
+import { push } from 'connected-react-router';
+
+export function* handlePatternsLoad({ payload }) {
+  try {
```

Comment:
Create and add **loading actions** in case of slow connection to inform user. Set it to `true` before fetching patterns and after successful load - set to `false`. In `catch` block also set it to `false`. Don't forget about `reducer` ;)

Reply:
done